pebbles/views/groups.py

```diff
@@ -138,18 +140,32 @@ def patch(self, group_id):
     @requires_admin
     def delete(self, group_id):
         group = Group.query.filter_by(id=group_id).first()
-        if group.name == 'System.default':
-            logging.warn("cannot delete the default system group")
-            return {"error": "Cannot delete the default system group"}, 422
+
         if not group:
             logging.warn("trying to delete non-existing group")
             abort(404)
-        blueprint_instances = group.blueprints.all()
-        for blueprint in blueprint_instances:
-            blueprint.current_status = 'deleted'
-        db.session.commit()
-        db.session.delete(group)
-        db.session.commit()
+        if group.name == 'System.default':
+            logging.warn("cannot delete the default system group")
+            return {"error": "Cannot delete the default system group"}, 422
+
+        group_blueprints = group.blueprints.all()
+
+        if not group_blueprints:
+            db.session.delete(group)
+            db.session.commit()
+        else:
+            for group_blueprint in group_blueprints:
```

Comment:
I realized you have used almost the same kind of code here: https://github.com/CSCfi/pebbles/commit/a6aef41d727a001c751fffa8ac68b8be2e43d848#diff-3633ba1bd0584c7e45c1688cbd74823eR173

Which is basically deletion of instances, I would recommend keeping the deletion code of an instance in a function somewhere which you can read if needed again